dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -292,6 +300,14 @@ public abstract class Entry<S extends Entry, T extends Version> implements Compa
     @JsonIgnore
     private List<Category> categories = new ArrayList<>();
 
+    @Transient
```

Comment:
Feels a bit like lazy loading without using JPA/Hibernate  lazy loading. 
Taking a quick look, I don't unfortunately see a way of defining collections via queries.

Reply:
> Feels a bit like lazy loading without using JPA/Hibernate lazy loading. Taking a quick look, I don't unfortunately see a way of defining collections via queries.

Yeah, I came up with the basic mechanism (populating a `Transient` field via `PopulateEntryListener`) when I did the categories work, way back when.  I couldn't find a better way, either.  If all of the calculated information is used only by `ElasticListener`, I would've put the code there.  But, as I vaguely recall, there were issues getting necessary DAOs into `ElasticListener`, and thus, the query is bolted on via the `Transient` and ``PopulateEntryListener`.  Possibly, now, given my much greater relative experience with the codebase, it would be an easier refactor...

Reply:
> > Feels a bit like lazy loading without using JPA/Hibernate lazy loading. Taking a quick look, I don't unfortunately see a way of defining collections via queries.
> 
> Yeah, I came up with the basic mechanism (populating a `Transient` field via `PopulateEntryListener`) when I did the categories work, way back when. I couldn't find a better way, either. If all of the calculated information is used only by `ElasticListener`, I would've put the code there. But, as I vaguely recall, there were issues getting necessary DAOs into `ElasticListener`, and thus, the query is bolted on via the `Transient` and `PopulateEntryListener`. Possibly, now, given my much greater relative experience with the codebase, it would be an easier refactor...

Agreed, I was initially going to inject the DAO(s) into ElasticListener, which did seem a tad messy and might break the unit tests, when I discovered the existing pattern in PopulateEntryListener..